forc-test/src/lib.rs

```diff
@@ -352,16 +450,43 @@ impl BuiltTests {
 /// First builds the package or workspace, ready for execution.
 pub fn build(opts: Opts) -> anyhow::Result<BuiltTests> {
     let build_opts = opts.into_build_opts();
+    let build_plan = pkg::BuildPlan::from_build_opts(&build_opts)?;
     let built = pkg::build_with_options(build_opts)?;
-    BuiltTests::from_built(built)
+    let built_members: HashMap<&pkg::Pinned, Arc<BuiltPackage>> = built.into_members().collect();
+
+    // For each member node collect their contract dependencies.
+    let member_contract_dependencies: HashMap<pkg::Pinned, Vec<Arc<pkg::BuiltPackage>>> =
+        build_plan
+            .member_nodes()
+            .map(|member_node| {
+                let graph = build_plan.graph();
+                let pinned_member = graph[member_node].clone();
+                let pinned_contract_dependencies: Vec<pkg::Pinned> = build_plan
+                    .contract_dependencies(member_node)
+                    .map(|contract_depency_node_ix| graph[contract_depency_node_ix].clone())
+                    .collect();
+                let contract_dependencies = pinned_contract_dependencies
+                    .iter()
```

Comment:
```suggestion
```
It looks like we might not need to collect the `pinned_contract_dependencies`, and can `filter_map` in this same iteration?